File: RealInfoUI/src/app/_models/propertySearch.ts
import { inherits } from "util";

export class searchBase{
    schema: string;
    constructor() {       
        this.schema ="";
    }
}

export class PropertySearchOwnerName extends searchBase {
    FirstName: string;
    FirstNameCondition: string;
    LastName: string;
    LastNameCondition: string;
    

    constructor() {
        super();
        this.FirstName = "";
        this.FirstNameCondition = "";
        this.LastName = "";
        this.LastNameCondition = "";
       
    }
}

export class PropertySearchTax extends searchBase{
    TaxNumber: string;

    constructor() {
        super();
        this.TaxNumber = "";
    }
}

export class PropertySearchAddress extends searchBase {
    StreetNumber: string;

    StreetName: string;

    City: string;

    ZipCode: string;

    constructor() {
        super();
        
        this.StreetNumber = "";

        this.StreetName = "";
        this.City = "";
        this.ZipCode = "";
    }

}

File: RealInfoViewModel/ViewModels/PropertySearch.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RealInfoViewModel.ViewModels
{
    public class PropertySearch{

        public string schema { get; set; }
    }
    public class PropertySearchOwnerName : PropertySearch
    {
        public string FirstName { get; set; }
        public string FirstNameCondition { get; set; }
        public string LastName { get; set; }
        public string LastNameCondition { get; set; }
      
    }

    public class PropertySearchTax : PropertySearch
    {
        public string TaxNumber { get; set; }
    }

    public class PropertySearchAddress : PropertySearch
    {
        public string StreetNumber { get; set; }

        public string StreetName { get; set; }

        public string City { get; set; }

        public string ZipCode { get; set; }


    }
}
